packages/kbn-alerts-ui-shared/src/rule_form/rule_definition/rule_alert_delay.test.tsx

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { RuleAlertDelay } from './rule_alert_delay';
+
+const mockOnChange = jest.fn();
+
+describe('RuleAlertDelay', () => {
+  test('Renders correctly', () => {
+    render(
+      <RuleAlertDelay
+        alertDelay={{
+          active: 5,
+        }}
+        onChange={mockOnChange}
+      />
+    );
+
+    expect(screen.getByTestId('alertDelay')).toBeInTheDocument();
+  });
+
+  test('Should handle input change', () => {
+    render(
+      <RuleAlertDelay
+        alertDelay={{
+          active: 5,
+        }}
+        onChange={mockOnChange}
+      />
+    );
+
+    fireEvent.change(screen.getByTestId('alertDelayInput'), {
```

Comment:
`userEvent` doesn't work because the EuiInputs keeps internal state

Reply:
`userEvent` uses `fireEvent` internally. Why does it matter if the EuiInputs keep an internal state?

Reply:
so `fireEvent` you define the `event` object, so you do something like `fireEvent.click(element, { target: value: 'blah' })`. Which means you bypass any internal state

`userEvent` uses an api like `userEvent.type(element, 'ABC')`, so it's actually typing in the input. In this test case, the start is starting with an empty `value` prop. But because EUI input keeps internal state, it actually starts to store the inputs unless I modify the `value` prop. Which means i need to create a harness for this test and have a `useState`, to turn it into a controlled input.